redbot/core/core_commands.py

```diff
@@ -53,13 +53,9 @@ def __init__(self, bot):
     @commands.command(hidden=True)
     async def ping(self, ctx):
         """Pong."""
-        if ctx.guild is None or ctx.channel.permissions_for(ctx.guild.me).add_reactions:
-            try:
-                await ctx.message.add_reaction("\U0001f3d3")  # ping pong paddle
-            except discord.errors.NotFound:
-                # this occurs if the message was deleted because of instant command delete
-                pass 
-        else:
+        try:
+            await ctx.message.add_reaction("\U0001f3d3")  # ping pong paddle
+        except [discord.errors.NotFound, discord.errors.Forbidden]:
```

Comment:
Not a fan of the try/except forbidden since we have the channel object, we shouldnt make avoidable bad API calls

Reply:
There's no way to know if the message still exists. I can't do something like `if message is None`, since, even if it got deleted, the message object is static.